### ISSUE TITLE
Custom Javascript

### ISSUE THREAD

**Issue by kaylee-98** [goauthentik/authentik#8402]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to set different CSS per tenant and sometimes even per flow. I would also like to further customise the functionality of certain things for my specific application. 

**Describe the solution you'd like**
I think it would be pretty straightforward to inject a custom javascript file just like the optional custom css file that gets injected on every page. This would take almost no work to implement and would open up basically infinite potential for customisation.

**Describe alternatives you've considered**
I thought about just adding this to my own install but decided against it as it would complicate updates.

**Comment by BeryJu** [comment]
this will be possible with #4748

**Comment by ewen-lbh** [comment]
If you're looking for a workaround and using kubernetes with the NGiNX ingress, you can do this:

0. Enable snippet annotations on the ingress controller, for example with `allowSnippetAnnotations: true` in NGiNX's helm chart values
1. Define a custom.js in your [configmap alongside `custom.css`](https://docs.goauthentik.io/docs/interfaces/flow/customization#custom-css)
2. Mount it to /web/dist/custom.js, just like custom.css
3. Add this to `server: ingress: annotations`:
   ```yaml
	nginx.ingress.kubernetes.io/configuration-snippet: |
            sub_filter "</head>" "<script src=/static/dist/custom.js></script></head>";
            proxy_set_header Accept-Encoding "";
    ```

**Comment by fentas** [comment]
Yes. With this simple change, we could solve all other open issues, like a show password functionality for a password recovery stage (prompt) and many other thing we have to make complicated expressions.

This feature is quite underrated, and it's surprising that there is no more upvoting for it.
Especially if there is a PR which would add this going into year 3.

**Comment by davidfregolim** [comment]
For the time being, I have landed here:

## Injecting Custom JS into Authentik

Authentik serves its own UI — we don't control its HTML. To inject a custom `<script>` tag into every Authentik page, we build a thin image layer on top of the official one.

## How it works

Authentik's pages (`admin.html`, `user.html`, `flow.html`) all extend a single Django base template: `/authentik/core/templates/base/skeleton.html`.

We use a Dockerfile that runs `sed` on the original template at build time, injecting a `<script>` tag just before `</body>`:

```dockerfile
ARG AK_VERSION=2026.2
FROM ghcr.io/goauthentik/server:${AK_VERSION}

USER root
RUN sed -i 's|</body>|    <script src="https://example.com/my-script.js"></script>\n    </body>|' \
    /authentik/core/templates/base/skeleton.html
USER 1000
```

Both `authentik-server` and `authentik-worker` use this image via `build:` in `compose.yaml` (instead of `image:`).

## Upgrading Authentik

Change `AK_VERSION` in the Dockerfile and rebuild. The `sed` runs against whatever `skeleton.html` the new image ships — no template copy to keep in sync.
